wagtail/contrib/search_promotions/admin_urls.py

```diff
@@ -4,7 +4,7 @@
 
 app_name = "wagtailsearchpromotions"
 urlpatterns = [
-    path("", views.index, name="index"),
+    path("", views.IndexView.as_view(), name="index"),
```

Comment:
@rohitsrma add another view for the results here, i.e.

```suggestion
    path("results/", views.IndexView.as_view(results_only=True), name="index_results"),
```